fedimint-core/src/encoding/mod.rs

```diff
@@ -1012,6 +1012,80 @@ impl<R: std::io::Read> bitcoin_io::Read for SimpleBitcoinRead<R> {
     }
 }
 
+/// Wrap buffering support for implementations of Read.
+/// A reader which keeps an internal buffer to avoid hitting the underlying
+/// stream directly for every read.
+///
+/// In order to avoid reading bytes past the first object, and those bytes then
+/// ending up getting dropped, this BufBitcoinReader operates in
+/// one-byte-increments.
```

Comment:
Worth pointing to the source where we copied it from.

Reply:
Done